crates/starknet-os/src/hints/output.rs

```diff
@@ -144,6 +162,58 @@ pub fn set_state_updates_start(
     Ok(())
 }
 
+pub const SET_COMPRESSED_START: &str = indoc! {r#"if use_kzg_da:
+    ids.compressed_start = segments.add()
+else:
+    # Assign a temporary segment, to be relocated into the output segment.
+    ids.compressed_start = segments.add_temp_segment()"#};
+
+pub fn set_compressed_start(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let use_kzg_da_felt = exec_scopes.get::<Felt252>(vars::scopes::USE_KZG_DA)?;
+
+    let use_kzg_da = if use_kzg_da_felt == Felt252::ONE {
+        Ok(true)
+    } else if use_kzg_da_felt == Felt252::ZERO {
+        Ok(false)
+    } else {
+        Err(HintError::CustomHint("ids.use_kzg_da is not a boolean".to_string().into_boxed_str()))
+    }?;
+
+    if use_kzg_da {
+        insert_value_from_var_name(vars::ids::COMPRESSED_START, vm.add_memory_segment(), vm, ids_data, ap_tracking)?;
+    } else {
+        // Assign a temporary segment, to be relocated into the output segment.
+        insert_value_from_var_name(vars::ids::COMPRESSED_START, vm.add_temporary_segment(), vm, ids_data, ap_tracking)?;
+    }
+
+    Ok(())
+}
+
+pub const SET_N_UPDATES_SMALL: &str =
+    indoc! {r#"ids.is_n_updates_small = ids.n_actual_updates < ids.N_UPDATES_SMALL_PACKING_BOUND"#};
+
+pub fn set_n_updates_small(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let n_actual_updates = get_integer_from_var_name(vars::ids::N_ACTUAL_UPDATES, vm, ids_data, ap_tracking)?;
+    let n_updates_small_packing_bound = get_constant(vars::ids::N_UPDATES_SMALL_PACKING_BOUND, constants)?;
+
+    let is_n_updates_small =
+        if n_actual_updates < *n_updates_small_packing_bound { Felt252::ZERO } else { Felt252::ONE };
```

Comment:
This should be the other way around
```suggestion
        if n_actual_updates < *n_updates_small_packing_bound { Felt252::ONE } else { Felt252::ZERO };
```

Reply:
Good catch! Fixed this one and the deserialize_output that was comparing the wrong value 💪